### PR TITLE
Add Google currency data provider.

### PR DESCRIPTION
**Pull request by rossigee**
As per subject, this patch adds a new provider based on the Google Currency Converter.

This was done for use with 9.0, and I'm just putting out there for anyone that might find it useful. I will submit a separate PR once I've updated and tested it to work against 10.0.

### CODE REVIEW

**Comment by adrienpeiffer** [comment]
@rossigee Can you fix pylint issues ?

**Comment by rossigee** [comment]
@adrienpeiffer - I've fixed the missing line it reports in the Python file. The RST file error doesn't appear to be related to my patch, but if it is, I don't know why or how to fix.

File: currency_rate_update/services/update_service_GOOGLE.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
```
Comment (adrienpeiffer) [inline currency_rate_update/services/update_service_GOOGLE.py]: Can you use a shorter header ?
e.g. https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py

**Review by adrienpeiffer — CHANGES_REQUESTED** [review]

File: currency_rate_update/services/update_service_GOOGLE.py (hunk 1 above)
Comment (rossigee) [reply currency_rate_update/services/update_service_GOOGLE.py]: Done.

**Comment by adrienpeiffer** [comment]
@rossigee I just tested your module on runbot. But why I try to update currency that takes a lot of time with no result. I tested only with one currency (GBP). Did I miss something ?

**Comment by rossigee** [comment]
@adrienpeiffer - Sorry, I'm not familiar with runbot. I have only tested this on real systems. How can I reproduce?

**Comment by adrienpeiffer** [comment]
The related runbot build is killed. I just started a new one. You have to wait the end of the build to test it

**Comment by rossigee** [comment]
@adrienpeiffer - I still do not understand. How can I reproduce the problem you are describing?

**Comment by adrienpeiffer** [comment]
You can test it here : http://3293799-481-6f94e7.runbot2.odoo-community.org/web/login

**Comment by rossigee** [comment]
@adrienpeiffer - what is the login?

**Comment by adrienpeiffer** [comment]
admin - admin :smile:

**Comment by rossigee** [comment]
@adrienpeiffer - thanks, that worked.

Unfortunately, I'm having trouble configuring the instance in order to test with it. This is what I did after logging in.

1. Enabled developer mode in the 'About' panel.
2. Enabled 'multi currency' in the Accounts -> Configuration -> Settings.
3. Checked there were active currencies in Accounts -> Multi-Currencies -> Currencies. Present are 'USD' and 'EUR', which should be fine.
3. Create a new currency rate update in Accounts -> Currency Rate Update.

At this point, regardless of whether I choose an existing webservice (i.e. Yahoo) or the new one, it does not offer me any currencies to add when I click 'Add an item' in the 'Currencies to update with this service' table.

What have I missed?

**Comment by adrienpeiffer** [comment]
All currencies was inactive. I just created a record of currency.rate.update.service in runbot

**Comment by rossigee** [comment]
OK, it seems you must choose the service, click 'Save' and *then* add currencies. I was trying to add currencies while creating the service, which it seems unable to do. That's a separate bug/issue, as is trying to add a currency without a service being selected, which produces a JS stacktrace.

Anyway. I can now see/reproduce the issue on runbot. However, I have no idea where to find runbot's instance logs to look for errors/warnings, or how to further debug the instance.

As a control measure, I have also added the Yahoo service and configured it similarly. This also exhibits the same behaviour, which suggests that either:

a) This is some kind of bug present in more than just the new service.
b) There is something about the runbot instance/configuration that is preventing this from working as expected.

Are we sure that the runbot instance is able to make outbound HTTPS calls successfully?

File: currency_rate_update/services/update_service_GOOGLE.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            self.validate_cur(curr)
+            if curr in self.invert_currencies:
+                res = self.get_url(url % (curr, main_currency))
+            else:
+                res = self.get_url(url % (main_currency, curr))
+            match = regexp.search(res)
+            if match:
+                val = match.group()[16:-11]
+                if curr in self.invert_currencies:
+                    val = str(1.0 / float(val))
```
Comment (elicoidal) [inline currency_rate_update/services/update_service_GOOGLE.py:60]: should not you test for val <>0?

**Review by elicoidal — CHANGES_REQUESTED** [review]
Thanks for the contribution!
Can you improve the README file:
* tags
* document title
* footer
* can you improve usage section with a procedure following comments https://github.com/OCA/account-financial-tools/pull/481#issuecomment-305794338 and https://github.com/OCA/account-financial-tools/pull/481#issuecomment-305803021

Can you bump version number in manifest?

File: currency_rate_update/services/update_service_GOOGLE.py (hunk 2 above)
Comment (rossigee) [reply currency_rate_update/services/update_service_GOOGLE.py:60]: I don't see why that would be necessary. The other service implementations don't check for zero. IMHO, if it were necessary, that should be the subject of a separate patch that addresses the potential issue in all the services.

**Comment by rossigee** [comment]
@elicoidal - This patch does not change any documented behaviour. Such improvements/additions to the README should be the subject of a separate patch.

**Comment by elicoidal** [comment]
@rossigee OK: understand your point about additional PR. 

Still a bump in version is necessary

**Comment by rossigee** [comment]
@elicoidal - the one in __openerp.py__? which of the five digits to bump?! I would guess '9.0.1.0.1'? (sorry, I'm new here!)

**Comment by elicoidal** [comment]
@rossigee yes following: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#version-numbers

**Comment by rossigee** [comment]
@elicoidal - thanks for that. It says the 'y' figure should be incremented for non-breaking new features (i.e. 9.0.1.1.0). I've updated the patch accordingly.

**Pull request closed without merging**